### ISSUE TITLE
Define cmd arguments and config options in one place.

### ISSUE THREAD

**Issue by real-yfprojects** [akaihola/darglint2#36]
At the moment arguments have to be 'defined' in multiple places for parsing from different configuration files and such. I saw people extending argparse to define config options in one place while parsing config files too.

I found [typer-config](https://github.com/maxb2/typer-config) a library for use with [typer](https://typer.tiangolo.com/) that might be a solution to this problem.

**Comment by akaihola** [comment]
[typer-config](https://github.com/maxb2/typer-config) is only one month old and a one-man project. [typer](https://typer.tiangolo.com/) does look like an established and well-maintained project and is based on [Click](https://click.palletsprojects.com/) which is even more so.

[typer-config](https://github.com/maxb2/typer-config) says it's inspired by [phha/click_config_file](https://github.com/phha/click_config_file) which seems like a not so active project as well (last commit a year ago, issue about [CI failure](/phha/click_config_file/issues/20) open for 3 years already). The README [mentions](/phha/click_config_file#why) there being "several existing implementations of config file support for Click", so maybe there's a well-established variant among those.

I'm writing this since I've been previously bitten by my decisions to introduce non-maintained dependencies for projects.

**Comment by real-yfprojects** [comment]
> I'm writing this since I've been previously bitten by my decisions to introduce non-maintained dependencies for projects.

I agree that one has to select dependencies carefully. The typer-config project was created in reaction to my participation in https://github.com/tiangolo/typer/issues/86#issuecomment-1529636251.
